scrapy/downloadermiddlewares/redirect.py

```diff
@@ -55,28 +55,25 @@ class RedirectMiddleware(BaseRedirectMiddleware):
 
     def process_response(self, request, response, spider):
         if (request.meta.get('dont_redirect', False) or
-               response.status in getattr(spider, 'handle_httpstatus_list', []) or
-               response.status in request.meta.get('handle_httpstatus_list', []) or
-               request.meta.get('handle_httpstatus_all', False)):
+                response.status in getattr(spider, 'handle_httpstatus_list', []) or
+                response.status in request.meta.get('handle_httpstatus_list', []) or
+                request.meta.get('handle_httpstatus_all', False)):
             return response
 
-        if request.method == 'HEAD':
-            if response.status in [301, 302, 303, 307] and 'Location' in response.headers:
-                redirected_url = urljoin(request.url, response.headers['location'])
+        location = None
+        if 'Location' in response.headers:
+            location = response.headers['location'].decode('latin1')
+
+        if location is not None and response.status in [301, 302, 303, 307]:
+            redirected_url = urljoin(request.url, location)
```

Comment:
In Python 2 `request.url` would be bytes, and urljoin wants bytes, but here `location` became unicode.
Do we have tests for non-ascii Location values?

It could also make sense to percent-encode location, but I'm not sure.